raocp/core/scenario_tree.py

```diff
@@ -106,22 +114,39 @@ def __make_ancestors_values_stages(self):
                 node_id = cursor + i
                 cover = self.__cover(int(values[node_id]))
                 length_cover = len(cover)
-                ancestors = np.concatenate((ancestors, node_id * np.ones((length_cover, ))))
+                ones = np.ones((length_cover, ), dtype=int)
+                ancestors = np.concatenate((ancestors, node_id * ones))
                 nodes_added_at_stage += length_cover
                 values = np.concatenate((values, cover))
             num_nodes_at_stage = nodes_added_at_stage
             cursor = cursor_new
-            stages = np.concatenate((stages, (1 + stage_idx) * np.ones(nodes_added_at_stage, )))
+            ones = np.ones(nodes_added_at_stage, dtype=int)
+            stages = np.concatenate((stages, (1 + stage_idx) * ones))
 
         for stage_idx in range(self.__stopping_time, self.__num_stages):
             ancestors = np.concatenate((ancestors, range(cursor, cursor+num_nodes_at_stage)))
             cursor += num_nodes_at_stage
-            stages = np.concatenate((stages, (1 + stage_idx) * np.ones(nodes_added_at_stage, )))
-            # TODO update `values` after stopping time?
+            ones = np.ones((nodes_added_at_stage,), dtype=int )
+            stages = np.concatenate((stages, (1 + stage_idx) * ones))
+            values = np.concatenate((values, values[-num_nodes_at_stage::]))
+
         return ancestors, values, stages
 
     def __make_probability_values(self):
```

Comment:
This implementation doesn't take into account the fact that there is a stopping time after which there is no more branching. To see this, add this line at the end of `test_markov_probability_of_node` in `tests/test_scenario_tree.py`:

```.py
self.assertAlmostEqual(0.005, tree.probability_of_node(20), delta=tol)
```